app/channels/stimulus_reflex/channel.rb

```diff
@@ -113,4 +117,10 @@ def fix_environment!
       controller.renderer.instance_variable_set(:@env, connection.env.merge(controller.renderer.instance_variable_get(:@env)))
     end
   end
+
+  def disable_full_debug
+    if StimulusReflex.config.debug == false && StimulusReflex.config.parent_channel == "ApplicationCable::Channel"
+      ActionCable.server.config.logger = Logger.new(nil)
```

Comment:
This line stops `ActionCable` from producing any verbose log messages. Note: this only works if a parent channel class is `ApplicationCable::Channel` of course.

Reply:
Actually, why is this important? I honestly don't follow. :)

Reply:
Well, since https://github.com/hopsoft/stimulus_reflex/pull/346 was merged we gained a possibility of using other channel class than `ApplicationCable::Channel`, didn't we? This if statement opts out of printing AC log messages only. Can we be sure that provided custom class is built on top of AC? If so, then we do not this check: `StimulusReflex.config.parent_channel == "ApplicationCable::Channel"`.

Reply:
At this juncture - in advance of potential future modular transports - SR cannot work if channel doesn't ultimately inherit from ActionCable. I think we should drop that guard.

I also admit that I didn't initially intend for SR to explicitly shut off the built-in ActionCable logging, just that it would be encouraged/possible alongside the functionality you're building.

See, the thing is that turning off AC logging also happens to make the whole system work faster, too! What's not to love, right?

Well, the problem is that the opposite of `StimulusReflex.config.debug == false` shouldn't be turning on AC debugging. Instead, I think we should just make strong recommendations to people that they should enable SR debugging and disable AC debugging, and show them how (as we do already, [here](https://docs.stimulusreflex.com/troubleshooting#server-side)).

Reply:
So, here's an interesting development. According to my testing, setting `ActionCable.server.config.logger = Logger.new(nil)` inside the channel doesn't have any impact until the 2nd connection is established... which makes sense because the channel is already live.

Anyhow, having sat on this for some hours, I think I'm going to pull out the `config.debug` stuff. The correct thing to do is provide good instructions, not impose our logic on how the developer wants to configure their Rails app. They could want both!

Reply:
_"The correct thing to do is provide good instructions, not impose our logic on how the developer wants to configure their Rails app. They could want both!" ❤️_

100% agree with it @leastbad